src/pool/srv_pool.c

```diff
@@ -1168,7 +1168,6 @@ static void
 fini_svc_pool(struct pool_svc *svc)
 {
 	D_ASSERT(svc->ps_pool != NULL);
-	ds_pool_iv_ns_update(svc->ps_pool, -1 /* master_rank */);
```

Comment:
why remove it? is it possible after fini_svc_pool() it receive some IV msg?

Reply:
During leader change, the new leader is being updated when it receives the pool map IV update. (in iv pre sync callback). in the mean time, this fini_svc_pool() will be called from pool svc step down, which might happen after pre_sync callback, then master will be set to -1.

So I remove this -1 set, so the old master will still sending stale IV rpc, and another nodes will reject it anyway.

Reply:
Sorry for not being as familiar with this part of the code (the pool IV namespace stuff). Not sure I understand the race that @wangdi1 desribed. But I wonder about this case

The step up callback invokes init_svc_pool() that sets the IV ns master rank to self. And if that step up callback function fails in its subsequent steps, it invokes fini_svc_pool() that (before this change) would reset the master rank to -1. But (with this change) would remain set to its own rank even in this error case. Is that also OK?

Reply:
> it invokes fini_svc_pool() that (before this change) would reset the master rank to -1. 

As long as if there is new leader step up, which will propagate the pool map to all ranks, where each rank will update its master rank in pre_sync callback. 

The issue is that without this change, if this fini_svc_pool()  happened after the pre_sync callback, triggered by the new leader, then  the master rank will be become -1 until the another leader switch.